### PR TITLE
[function] add trigonometric func

### PR DESCRIPTION
**Pull request by Junnplus**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add trigonometric func.

```
mysql> SELECT SIN(PI());
+------------------------------------+
| SIN(PI())                          |
+------------------------------------+
| 0.00000000000000012246467991473532 |
+------------------------------------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.000 sec., 2.46 thousand rows/sec., 2.46 KB/sec.

mysql> SELECT TAN(PI());
+-------------------------------------+
| TAN(PI())                           |
+-------------------------------------+
| -0.00000000000000012246467991473532 |
+-------------------------------------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.000 sec., 2.44 thousand rows/sec., 2.44 KB/sec.

mysql> SELECT TAN(PI()+1);
+--------------------+
| TAN((PI() + 1))    |
+--------------------+
| 1.5574077246549016 |
+--------------------+
1 row in set (0.00 sec)
Read 1 rows, 1 B in 0.001 sec., 1.3 thousand rows/sec., 1.3 KB/sec.

mysql> SELECT COS(PI());
+-----------+
| COS(PI()) |
+-----------+
|        -1 |
+-----------+
1 row in set (0.00 sec)
Read 1 rows, 1 B in 0.000 sec., 2.64 thousand rows/sec., 2.64 KB/sec.

mysql> SELECT COT(12);
+---------------------+
| COT(12)             |
+---------------------+
| -1.5726734063976895 |
+---------------------+
1 row in set (0.00 sec)
Read 1 rows, 1 B in 0.000 sec., 2.58 thousand rows/sec., 2.58 KB/sec.
```

## Changelog

- New Feature

## Related Issues

https://github.com/datafuselabs/databend/issues/2563

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datavalues/src/series/series_impl.rs [hunk 1]
```diff
@@ -154,7 +154,7 @@ pub trait SeriesTrait: Send + Sync + fmt::Debug {
     /// Unpack to DFArray of data_type u64
     fn u64(&self) -> Result<&DFUInt64Array> {
         Err(ErrorCode::IllegalDataType(format!(
-            "{:?} != u32",
+            "{:?} != u64",
```
Comment (sundy-li) [inline common/datavalues/src/series/series_impl.rs:157]: Great fix.

**Review by sundy-li — APPROVED** [review]
/LGTM

File: common/datavalues/src/series/series_impl.rs (hunk 1 above)
Comment (BohuTANG) [reply common/datavalues/src/series/series_impl.rs:157]: 👍

**Pull request merged**